scripts/get_arch_and_platform.sh

```diff
@@ -0,0 +1,26 @@
+if command -v uname &> /dev/null
+then
+  ARCH=$(uname -m)
+elif command -v dpkg &> /dev/null
+then
+  ARCH=$(dpkg --print-architecture)
+elif command -v apk &> /dev/null
+then
+  ARCH=$(apk --print-arch)
+else
+  echo "Could not detect system architecture. Please specify with ARCH env variable"
+  exit 1
+fi
+
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Linux*)     PLATFORM=linux;;
+    Darwin*)    PLATFORM=mac;;
+    CYGWIN*)    PLATFORM=cygwin;;
+    MINGW*)     PLATFORM=mingw;;
```

Comment:
when do we expect `cygwin`, `mingw` cases to be hit?

Reply:
they seems to be some unix like shells for windows, but I didn't test them.